Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -657,7 +657,8 @@ RefPtr<FunctionExpression> Parser::try_parse_arrow_function_expression(bool expe
 
     return create_ast_node<FunctionExpression>(
         { m_state.current_token.filename(), rule_start.position(), position() }, "", move(body),
-        move(parameters), function_length, FunctionKind::Regular, body->in_strict_mode(), true);
+        move(parameters), function_length, FunctionKind::Regular, body->in_strict_mode(),
+        /* might_need_arguments_object */ false, /* is_arrow_function */ true);
```

Comment:
Doesn't have to be in this PR, but YES/NO class enums seem perfect for here

Reply:
I think it'd be useful to have something like `FunctionNode::Flags` instead, then we can put `IsStrict`, `IsArrowFunction`, ... in there all at once.
Definitely meant to be temporary though, yep.

Reply:
Yeah, bit flags would also help with all the repeated TemporaryChance statements